addons/l10n_vn_edi/wizard/account_move_send.py

```diff
@@ -0,0 +1,167 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import json
+
+from odoo import api, fields, models, SUPERUSER_ID, _
+
+
+class AccountMoveSendSInvoice(models.TransientModel):
+    _inherit = 'account.move.send'
+
+    l10n_vn_edi_enable = fields.Boolean(
+        compute='_compute_l10n_vn_edi_enable',
+    )
+    l10n_vn_edi_send_checkbox = fields.Boolean(
+        compute='_compute_l10n_vn_edi_send_checkbox',
+        string='Send to SInvoice',
+        readonly=False,
+        store=True,
+    )
+    l10n_vn_edi_generate_file_checkbox = fields.Boolean(
+        compute='_compute_l10n_vn_edi_generate_file_checkbox',
+        string='Generate SInvoice file',
+        readonly=False,
+        store=True,
+    )
+
+    def _get_wizard_values(self):
+        # EXTENDS 'account'
+        values = super()._get_wizard_values()
+        values['l10n_vn_edi_send'] = self.l10n_vn_edi_send_checkbox
+        values['l10n_vn_edi_generate'] = self.l10n_vn_edi_generate_file_checkbox
+        return values
+
+    @api.depends('move_ids')
+    def _compute_l10n_vn_edi_enable(self):
+        for wizard in self:
+            wizard.l10n_vn_edi_enable = any(move.l10n_vn_edi_state == 'ready_to_send' for move in wizard.move_ids)
+
+    @api.depends('l10n_vn_edi_enable')
+    def _compute_l10n_vn_edi_generate_file_checkbox(self):
+        # E-invoicing is a legal requirement in Vietnam, so we can enable by default.
+        for wizard in self:
+            wizard.l10n_vn_edi_generate_file_checkbox = wizard.l10n_vn_edi_enable
+
+    @api.depends('l10n_vn_edi_enable')
+    def _compute_l10n_vn_edi_send_checkbox(self):
+        # E-invoicing is a legal requirement in Vietnam, so we can enable by default.
+        for wizard in self:
+            wizard.l10n_vn_edi_send_checkbox = wizard.l10n_vn_edi_enable
+
+    # -------------------------------------------------------------------------
+    # ATTACHMENTS
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _get_invoice_extra_attachments(self, move):
+        # EXTENDS 'account'
+        return super()._get_invoice_extra_attachments(move) + move.l10n_vn_edi_sinvoice_file_id
+
+    @api.model
+    def _get_invoice_extra_attachments_data(self, move):
+        # EXTENDS 'account'
+        vals = super()._get_invoice_extra_attachments_data(move)
+
+        # Ensure that we never send the sinvoice file by mail, but we still want to allow it to be downloaded.
+        for val in vals:
+            if val['id'] == move.l10n_vn_edi_sinvoice_file_id.id:
```

Comment:
Ah i see here you skip it, but i think we need to allow option so send pdf (the one get from sinvoice) to client. Also, i think you miss an very important file which is an xml (this one also provided by sinvoice). This xml file i'm talking about is a legal file in Vietnam , you should also send that to customer (or course if we have some option for that it would be nice)
https://newca.vn/hoa-don-dien-tu-luu-file-pdf-hay-xml/ . Sorry again no english found :(
![image](https://github.com/odoo/odoo/assets/56789189/8de79787-b2b1-4acd-b78c-6c1aee5880b3)

Reply:
Thank you, I will double check about this flow with our Product Owner and see his opinion on this subject :)

Reply:
Hi @duong77476-viindoo ! Just for you information, I talked about this and the product owner agreed on the need.
I will work on adding that feature (Pdf + XML sent with the email, and also attached automatically to the invoice) next week :)

Reply:
Nice, glad i can help

Reply:
Thanks! Would the xsl file also be required? Or can I omit it?

Reply:
@vin-odoo you mean the xml file which we will obtain from sinvoice ?

Reply:
@duong77476-viindoo When downloading the xml file, it comes with a xsl file. (See both attached as txt since github doesn't want xml/xsl files)
I was wondering if this is useful in any way or if I can ignore that file. 
[0100109106-509-K24TUT102.xml.txt](https://github.com/odoo/odoo/files/15057230/0100109106-509-K24TUT102.xml.txt)
[invoice.xsl.txt](https://github.com/odoo/odoo/files/15057231/invoice.xsl.txt)

Reply:
oh yes i see, i think we don't need the `xsl` file. My company normally will send `representation file` (the pdf file obtain from sinvoice) and `xml` file (actually we send the whole zip which contain both xml and xsl to client because i'm too lazy to use python code to extract them hehe)

Reply:
And the accountant doesn't argue anything so basically i send pdf and zip file to client :v